### PR TITLE
feat: on-ramp campaign updates for mercuryo

### PR DESCRIPTION
**Pull request by ChefBingbong**
### Summary
🎨🔄🗑️

Refactored the fiat on-ramp feature to use the Mercuryo widget instead of the Moonpay widget. Removed unused code and logic related to the new customer and provider fee features. Improved the UI and UX of the buy crypto feature and the CryptoCard component. Added translation support for the buy crypto feature.

> _`Moonpay` fades away_
> _`Mercuryo` shines in spring_
> _`buyCrypto` simpler_

### Walkthrough
*  Update `CryptoCard` component to use different background color based on theme and click state ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-44ba08619785c9500e8a2cb2dd4f1711ac4cc76654e97e0f890392043c8ddcf9L33-R34))
*  Replace `MOONPAY_SIGN_URL` constant with `ONRAMP_API_BASE_URL` constant in `FiatOnRampModal` component and `endpoints` config file ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L16-R16), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L112-R118), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-7238357db46da6bb48ca41841457ac0e832ee21fb8fd76a6b3c9dbf701ce8c62L14-R15), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-b5de45ec9907f720376de68df6bc2243339f1486c26645f69d767f54266afe42L1-R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-a427ea9a864cb04db3a1982186a0b9cde2fee509a2c7cb5ba85072f023eee734L1-R6))
*  Import and use `crypto` module to generate random strings for transaction ids in `FiatOnRampModal` component ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905R33), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905R91-R95), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905R262), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905R279))
*  Add `isTestEnv` query parameter to moonpay signed url request in `FiatOnRampModal` component ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905R133))
*  Remove `onIsNewCustomer` action handler and related logic from `FiatOnRampModal` component and `buyCrypto` state files ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L195), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L203-R209), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-75772925913a4b43e6e0b041906b4443e30f351ceb79d0db3297928e18f1d7c3L9), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-75772925913a4b43e6e0b041906b4443e30f351ceb79d0db3297928e18f1d7c3L20), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-75772925913a4b43e6e0b041906b4443e30f351ceb79d0db3297928e18f1d7c3L31), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L20-R20), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L230), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L275-L283), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L289), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L319), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L361-L369), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L382), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5L12-R15), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5L34), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5L76-L78), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5L93), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5L110))
*  Add `MERCURYO_WIDGET_ID` and `MERCURYO_WIDGET_URL` constants to `endpoints` config file and use them in `FiatOnRampModal` component ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-7238357db46da6bb48ca41841457ac0e832ee21fb8fd76a6b3c9dbf701ce8c62L91-R95), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-ec1dc15e7eaff4dffe6bf7d046f9b9f4c26a27a63089eee9af39f63659bb0905L336-R335))
*  Add `SUPPORT_BUY_CRYPTO` constant to `WalletInfo` component and use it to conditionally show buy crypto tooltip and icon for zero balance of BNB or native token ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-667fa9ffb5f38ba848d89c70e2b060c60ce67fee0185391f612eb425169911beR33), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-667fa9ffb5f38ba848d89c70e2b060c60ce67fee0185391f612eb425169911beR94-R96), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-667fa9ffb5f38ba848d89c70e2b060c60ce67fee0185391f612eb425169911beL102-R106), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-667fa9ffb5f38ba848d89c70e2b060c60ce67fee0185391f612eb425169911beL140-R162), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-667fa9ffb5f38ba848d89c70e2b060c60ce67fee0185391f612eb425169911beL179-R215))
*  Import and use `ONRAMP_PROVIDERS` constant in `Accordion` component and sort `combinedQuotes` array by putting mercuryo quotes at the end ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-253a295a5158be14f93bb2babc36632f771b44154ab7affa2405b5f9863b0199R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-253a295a5158be14f93bb2babc36632f771b44154ab7affa2405b5f9863b0199R20-R25))
*  Update `AccordionItem` component to remove unused `useBuyCryptoState` hook and related variables, add translation for "Quote not available" text, remove logic for showing moonpay campaign, and add logic for showing mercuryo campaign ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L13), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L62-R63), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L111-R108), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L137-R134), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L148-R145), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L156-R157))
*  Update `usePriceQuotes` hook to remove unused `isNewCustomer` variable and related logic, and simplify `sortProviderQuotes` function ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-05cf008d35fa21dfccb294f5fe1bb9a2bc335765a3dab53a36b8b54ff7944e72L19-R22), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-05cf008d35fa21dfccb294f5fe1bb9a2bc335765a3dab53a36b8b54ff7944e72L32-R34), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-05cf008d35fa21dfccb294f5fe1bb9a2bc335765a3dab53a36b8b54ff7944e72L58-R55), [link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-05cf008d35fa21dfccb294f5fe1bb9a2bc335765a3dab53a36b8b54ff7944e72L64-R61))
*  Add translation key and value for "Quote not available" text to `translations` file ([link](https://github.com/pancakeswap/pancake-frontend/pull/7667/files?diff=unified&w=0#diff-eb2ab983e2cefc22516cb7814c86346f4b2bd2b971980952868173095d48f459L2672-R2673))

### CODE REVIEW

File: apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx [hunk 1]
```diff
@@ -10,15 +10,16 @@ import {
...
   Text,
   useModal,
-  Liquidity,
 } from '@pancakeswap/uikit'
+import { LoadingDot } from '@pancakeswap/uikit/widgets/Liquidity'
+
 import { CommitButton } from 'components/CommitButton'
-import { MERCURYO_WIDGET_ID, MOONPAY_SIGN_URL, ONRAMP_API_BASE_URL } from 'config/constants/endpoints'
+import { MERCURYO_WIDGET_ID, MERCURYO_WIDGET_URL, ONRAMP_API_BASE_URL } from 'config/constants/endpoints'
+import crypto from 'crypto'
```
Comment (chefjackson) [inline apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx:18]: I think this package is deprecated?

Comment (chefjackson) [reply apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx:18]: should be a built-in nodejs module. If we're using on fe we need to find something else. Let's use nanoid if you want to generate random string?

File: apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx [hunk 2]
```diff
@@ -87,11 +88,16 @@ interface FetchResponse {
   urlWithSignature: string
 }
 
+function generateRandomString(length) {
+  const randomBytes = crypto.randomBytes(length)
```
Comment (chefjackson) [inline apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx:92]: maybe crypto instance?
https://developer.mozilla.org/en-US/docs/Web/API/crypto_property

File: apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx [hunk 3]
```diff
@@ -10,15 +10,16 @@ import {
...
 import { CommitButton } from 'components/CommitButton'
-import { MERCURYO_WIDGET_ID, MOONPAY_SIGN_URL, ONRAMP_API_BASE_URL } from 'config/constants/endpoints'
+import { MERCURYO_WIDGET_ID, MERCURYO_WIDGET_URL, ONRAMP_API_BASE_URL } from 'config/constants/endpoints'
+import crypto from 'crypto'
 import { useActiveChainId } from 'hooks/useActiveChainId'
 import Script from 'next/script'
 import { Dispatch, ReactNode, SetStateAction, memo, useCallback, useEffect, useState } from 'react'
-import { useBuyCryptoActionHandlers } from 'state/buyCrypto/hooks'
-import { styled, useTheme } from 'styled-components'
+import styled, { useTheme } from 'styled-components'
```
[the remark this answers is not in the document]
Comment (ChefBingbong) [reply apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx:22]: didnt know this i will change

File: apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx (hunk 2 above)
Comment (ChefBingbong) [reply apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx:92]: no just need to generate random string. will actually use uuid. just orgnlly used crypto to save the exttra package

Comment (ChefBingbong) [reply apps/web/src/components/FiatOnRampModal/FiatOnRampModal.tsx]: removed

File: apps/web/src/config/constants/endpoints.ts [hunk 4]
```diff
@@ -95,6 +94,12 @@ export const QUOTING_API = `${process.env.NEXT_PUBLIC_QUOTING_API}/v0/quote`
...
 export const FARMS_API = 'https://farms-api.pancakeswap.com'
 
-export const MERCURYO_WIDGET_ID = process.env.NEXT_PUBLIC_MERCURYO_WIDGET_ID || '76ba4ff5-2686-4ed4-8666-fadb0d9a5888'
+export const MERCURYO_WIDGET_ID = process.env.NEXT_PUBLIC_MERCURYO_WIDGET_ID || '64d1f9f9-85ee-4558-8168-1dc0e7057ce6'
 
 export const MOONPAY_API_KEY = process.env.NEXT_PUBLIC_MOONPAY_LIVE_KEY || 'pk_test_1Ibe44lMglFVL8COOYO7SEKnIBrzrp54'
+
+// no need for extra public env
+export const MERCURYO_WIDGET_URL =
+  typeof window !== 'undefined' && window.location.origin.includes('https')
```
[the remark this answers is not in the document]
Comment (ChefBingbong) [reply apps/web/src/config/constants/endpoints.ts]: changed

File: apps/web/src/views/BuyCrypto/components/AccordionDropdown/AccordionItem.tsx [hunk 5]
```diff
@@ -1,16 +1,15 @@
 import { Box, Flex, InfoIcon, RowBetween, Text, TooltipText, useTooltip } from '@pancakeswap/uikit'
 import { CryptoCard } from 'components/Card'
 import { FiatOnRampModalButton } from 'components/FiatOnRampModal/FiatOnRampModal'
-import { Dispatch, SetStateAction, useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { Dispatch, SetStateAction, useCallback, useEffect, useRef, useState } from 'react'
 import { getRefValue } from 'views/BuyCrypto/hooks/useGetRefValue'
 import { CryptoFormView, ProviderQuote } from 'views/BuyCrypto/types'
-import { styled } from 'styled-components'
+import styled from 'styled-components'
```
[the remark this answers is not in the document]
Comment (ChefBingbong) [reply apps/web/src/views/BuyCrypto/components/AccordionDropdown/AccordionItem.tsx:7]: resolved

**Review by chefjackson — APPROVED** [review]

**Pull request merged**